core/vnl/vnl_vector.h

```diff
@@ -439,33 +474,42 @@ ::put(size_t i, T const& v)
 // \relatesalso vnl_vector
 // \relatesalso vnl_matrix
 template<class T>
-inline vnl_vector<T> operator*(vnl_matrix<T> const& m, vnl_vector<T> const& v)
+inline vnl_vector<T> operator*(vnl_matrix<T> const& M, vnl_vector<T> const& v)
 {
-  return vnl_vector<T>(m, v, vnl_tag_mul());
+  vnl_vector<T> result(M.rows());
+#ifndef NDEBUG
+  if (M.cols() != v.size())
+    vnl_error_vector_dimension ("vnl_vector<>::operator*(M, v)", M.cols(), v.size());
+#endif
+  vnl_sse<T>::matrix_x_vector(M.begin(), v.begin(), result.begin(), M.rows(), M.cols());
+  return result;
 }
 
 //: add scalar and vector. O(n).
 // \relatesalso vnl_vector
 template<class T>
 inline vnl_vector<T> operator+(T s, vnl_vector<T> const& v)
 {
-  return vnl_vector<T>(v, s, vnl_tag_add());
+  return v.operator+(s);
 }
 
 //: subtract vector from scalar. O(n).
 // \relatesalso vnl_vector
 template<class T>
 inline vnl_vector<T> operator-(T s, vnl_vector<T> const& v)
 {
-  return vnl_vector<T>(-v, s, vnl_tag_add());
+  vnl_vector<T> result(v.size());
+  for(size_t i=0; i< result.size(); ++i)
+    result[i]= s-1*v[i];
```

Comment:
Why is `1` required here? Isn't `result[i] = s-v[i];` sufficient?

Reply:
Fixed, but forgot missed adding the patch.  Will add now.